### DISCUSSION TITLE
Renovate will suggest updates to bicep that are not yet supported by the CLI

### DISCUSSION THREAD

**Discussion by johan-lindqvist** [renovatebot/renovate#21104] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

Never worked

### Wanted end result.

Renovate should not suggest updates to Bicep types before they're supported by the Azure Bicep CLI.

### What you tried so far.

Ran az bicep build on the latest update of bicep types that renovate suggested but get errors that the types don't exist.

Since there seems to be a lag between when the types repo (https://github.com/Azure/bicep-types-az/blob/main/generated/index.json) is updated and when the Azure Bicep CLI is updated, renovate suggests updates that won't work. I am aware that we can have checks in our repositories that run az bicep build, but it doesn't seem great that renovate is suggesting updates that are guaranteed to not work. Maybe renovate could somehow check if the types have been added to the Azure Bicep CLI before suggesting changes.

**Comment by rarkins** [comment]
@skyfrk @JamieMagee do we have any way to know authoritatively when a release is available to the CLI? If not, is there a value of `stabilityDays` which we could apply by default for `bicep` which would work around the problem reasonably reliably?

**Reply by JamieMagee** [reply]
The authoritative source for the types is the [`Azure.Bicep.Types.Az`][1] package that is generated from the JSON in the `bicep-types-az` repository.

A simple fix would be to set `stabilityDays` to `30 days`.

A proper fix would be to:
1. Find the most recent publish date of the `Azure.Bicep.Types.Az` NuGet package
2. Use the GitHub [list commits][2] API with the `until` parameter set to the most recent publish date of the NuGet package
3. Pick the commit closest to the publish date
4. Fetch from that commit, instead of `master`.

So for an example:
1. The most recent publish date of `Azure.Bicep.Types.Az` is [version `0.2.364` at 2023-03-02T22:48:36Z][3]
2. [Listing the commits][4] we see the most recent commit is [`21fcdd8d5b524f60e0471b2b1615a05a8082aae2`][5]
3. Which allows us to generate the following URL: https://raw.githubusercontent.com/Azure/bicep-types-az/21fcdd8d5b524f60e0471b2b1615a05a8082aae2/generated/index.json

[1]: https://www.nuget.org/packages/Azure.Bicep.Types.Az 
[2]: https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits
[3]: https://www.nuget.org/packages/Azure.Bicep.Types.Az/0.2.364
[4]: https://api.github.com/repos/azure/bicep-types-az/commits?until=2023-03-02T22:48:36Z&per_page=1
[5]: https://github.com/Azure/bicep-types-az/commit/21fcdd8d5b524f60e0471b2b1615a05a8082aae2

**Reply by johan-lindqvist** [reply]
Yeah, stability days to 30 was the temporary solution I used as well. Would be nice with the other more reliable solution @JamieMagee suggested. 👍

**Reply by HonkingGoose** [reply, marked as answer]
I created this issue to track the "simple fix" or "proper fix":

- https://github.com/renovatebot/renovate/issues/21452

**Reply by johan-lindqvist** [reply]
30 days of stability has passed and it still doesn't seem that the az bicep CLI has support. So it appears the simple fix was too optimistic 😒

**Reply by HonkingGoose** [reply]
We'll have to fix it the proper way then. 😄

**Reply by agra6475** [reply]
I assume this is never getting fixed at this point. :(
